src/App/adapter/init.js

```diff
@@ -19,11 +19,22 @@ const makeLabelCandidates = (obj) => {
   return { ...obj, label }
 }
 
-function init(filterables) {
-  const list = Object.entries(filterables).flatMap(([k, v]) =>
-    v.flatMap((i) => ({ ...i, group: k })),
-  )
-  return list.map(makeLabelCandidates).reduce(uniqueLabelReducer, [])
+function init(groupedFilterables) {
+  const filterablesArr = Object.entries(groupedFilterables)
+    .flatMap(([group, arr]) => arr.map((f) => ({ ...f, group }))) // flatten filterables, while keeping a reference to their groups
+    .map(makeLabelCandidates) // generate missing labels
+    .reduce(uniqueLabelReducer, []) // make sure labels are unique
+    // add specific operators
+    .map((obj) =>
+      obj.range
+        ? { ...obj, operator: 'between' }
+        : ['title', 'sample_chemical_formula'].includes(obj.name)
```

Comment:
I've moved this here, as it has to be done before the line below. Obviously, if we want the adapter to be generic, it should not have to know about `title` and `sample_chemical_formula`. I'm hoping we can add a property to the relevant `filterables` that we can use to identify this use case in the same way as `obj.range` above. I will work on this next.